### ISSUE TITLE
npm scripts as tasks out-of-the-box

### ISSUE THREAD

**Issue by kieferrm** [Microsoft/vscode#32163]
I created a new repo, set it up for TS with the following `package.json` file.

```json
{
    "name": "<name>",
    "version": "0.0.1",
    "dependencies": {
        "request": "2.81.0"
    },
    "devDependencies": {
        "mocha": "3.5.0",
        "typescript": "2.4.2",
        "@types/node": "8.0.19",
        "@types/mocha": "2.2.41",
        "@types/request": "2.0.0"
    },
    "engines": {
        "node": "7.9.0",
        "npm": "4.2.0"
    },
    "scripts": {
        "start": "node out/sample.js",
        "test": "mocha --reporter spec out/test",
        "watch": "tsc -watch"
    }
}
```
Shouldn't my npm scripts be autodetected by the task runner?

**Comment by dbaeumer** [comment]
Works for me. Any additional steps to reproduce:

<img width="683" alt="capture" src="https://user-images.githubusercontent.com/1931590/29113197-b246c87e-7cf0-11e7-9f78-1ffd6b759a6a.png">

**Comment by warpdesign** [comment]
@dbaeumer Is there any task to run to have vscode autodetect nom scripts or is it automatic? I have some tasks, but not every task defined in my package.json appear.

What happens in a workspace with multiples package.json files each with scripts?

**Comment by kieferrm** [comment]
@dbaeumer not sure it helps but here is what I just tried:
- `Tasks: Run Tasks` -> told me that there are no detected tasks
- `Tasks: Configure Task Runner`>`Other` -> created the sample echo task
- `Tasks: Run Tasks` -> the echo tasks and the npm tasks were detected
- deleted `.vscode/tasks.json`
- `Tasks: Run Tasks` -> npm tasks detected
- restart vscode with new empty user data dir
- `Tasks: Run Tasks` -> npm tasks detected
- deleted folder, recloned repo
- `Tasks: Run Tasks` -> npm tasks detected

So, now I'm in a state in which tasks detected works and it seems I can't back to the state I was in until I ran `Configure Task Runner`.

**Comment by dbaeumer** [comment]
@kieferrm  There might be one thing: npm auto detection is only available for tasks v2. So you either need to have no tasks.json or one with the version number `"version": "2.0.0"`. Task version "0.1.0" doesn't have auto detection support for npm since the extension host API came with task 2.0.

@warpdesign see my comment above.

**Comment by kieferrm** [comment]
@dbaeumer I figured the task 2.0 piece that's why I ran `Configure Task` runner. But then everything continued to work after deleting the `tasks.json` which is in sync with what you say above.

It seems there is a path in 1.15. that tricks the task system into thinking it's in version 0.1.0 although it doesn't have a  `tasks.json` file. But the only thing I remember I did in respect to tasks was `Run task` and didn't get any to choose from.

As I can't reproduce anymore, I'll close it.
